Moving unit-costing parameters to their own block

bknueven opened the pull request:
## Fixes/Resolves: Work towards #676 

## Summary/Motivation:
The WaterTAP costing package has too many parameters crowding the namespace. This puts the parameters for each unit on their own block.

## Changes proposed in this PR:
- Reorganize costing parameters in WaterTAP Costing Package
- Touch a few affected tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

Commit:
moving unit-costing parameters to their own block

andrewlee94 reviewed (commented):
One comment that we could put as a feature for the future, but looks good.

hunterbarber commented on watertap/costing/watertap_costing_package.py:
```diff
@@ -1140,10 +1232,10 @@ def cost_gac(blk):
-                        (1 - blk.costing_package.gac_regen_frac)
-                        * blk.unit_model.gac_mass_replace_rate
+                        (1 - blk.costing_package.gac.regen_frac)
+                        * blk.unit_model.gac.mass_replace_rate
```
```suggestion
                        * blk.unit_model.gac_mass_replace_rate
```

A unit_model variable with the prefix `gac_` got changed when find and replacing `costing.gac_` variables

hunterbarber committed:
Update watertap/costing/watertap_costing_package.py

A unit_model variable with the prefix gac_ got changed when find and replacing costing.gac_ variables

Commit:
fixing lsrro example

bknueven commented:
> One comment that we could put as a feature for the future, but looks good.

@andrewlee94 I'm not seeing another comment from you on this PR. Curious what your thoughts are.

In line with #676, my next thing to-do was to add the mechanism to construct these blocks on-demand.

andrewlee94 commented:
@bknueven I must have forgotten to submit the comment - I was just noting that I think we could easily set it up so that the unit-specific sub-methods are only triggered the first time a unit of that type is constructed. This would help keep down the number of unused parameters (and Blocks) required.

hunterbarber reviewed (approved):
Assuming the "add the mechanism to construct these blocks on-demand" will be in subsequent PRs, this first step of moving parameters looks good to me.

bknueven commented:
> Assuming the "add the mechanism to construct these blocks on-demand" will be in subsequent PRs, this first step of moving parameters looks good to me.

Yes, I will do that subsequently. This is ready to merge as-is, IMHO.

adam-a-a committed:
Merge branch 'main' into watertap_costing_improvements_1

Commit:
cleaning up merge

bknueven committed:
Merge branch 'main' into watertap_costing_improvements_1
